apps/drivers-service/src/drivers-service.service.ts

```diff
@@ -319,6 +328,122 @@ export class DriversServiceService {
     }
   }
 
+  async createFleetOrganization (payload: CreateAccountWithOrganizationDto): Promise<ResponseWithStatus> {
+    const existingOrganization = await this.fleetOrgRepository.findOne({
+      name: payload.organization.toLowerCase()
+    })
+
+    if (existingOrganization !== null) {
+      throw new FitRpcException(
+        'Organization name already already exists',
+        HttpStatus.CONFLICT
+      )
+    }
+
+    try {
+      const createOrganization = await this.fleetOrgRepository.create({
+        name: payload.organization,
+        inviteLink: RandomGen.genRandomString()
+      })
+
+      const newMember = await this.fleetMemberRepository.create({
+        firstName: payload.firstName,
+        lastName: payload.lastName,
+        phone: internationalisePhoneNumber(payload.phone),
+        email: payload.email,
+        password: await bcrypt.hash(payload.password, 10),
+        organization: createOrganization._id.toString()
+      })
+
+      await this.fleetOrgRepository.findOneAndUpdate({
+        _id: createOrganization._id
+      }, {
+        $push: { members: newMember._id, owners: newMember._id }
```

Comment:
Same as above. there will be a type cast error. All _ids should be converted go string

Reply:
Fixed @siradji